### PR TITLE
DOC-1602: Added note about regional by row tables and region changes

### PR DESCRIPTION
**Pull request by nickvigilante**
Fixes DOC-1602, DOC-1603

- Region changes to a DB cannot occur at the same time as index modifications or locality changes to `REGIONAL BY ROW` tables

### CODE REVIEW

File: _includes/v21.2/sql/indexes-regional-by-row.md [hunk 1]
```diff
@@ -1 +1,3 @@
  In [multi-region deployments](multiregion-overview.html), most users should use [`REGIONAL BY ROW` tables](multiregion-overview.html#regional-by-row-tables) instead of explicit index [partitioning](partitioning.html). When you add an index to a `REGIONAL BY ROW` table, it is automatically partitioned on the [`crdb_region` column](set-locality.html#crdb_region). Explicit index partitioning is not required.
+
+While CockroachDB process an [`ADD REGION`](add-region.html) or [`DROP REGION`](drop-region.html) statement on a particular database, creating or modifying an index will throw an error. Similarly, all [`ADD REGION`](add-region.html) and [`DROP REGION`](drop-region.html) statements will be blocked an index is being modified on a `REGIONAL BY ROW` table within the same database.
```
Comment (gemma-shay) [inline _includes/v21.2/sql/indexes-regional-by-row.md:3]:
```suggestion
While CockroachDB processes an [`ADD REGION`](add-region.html) or [`DROP REGION`](drop-region.html) statement on a particular database, creating or modifying an index will throw an error. Similarly, all [`ADD REGION`](add-region.html) and [`DROP REGION`](drop-region.html) statements will be blocked if an index is being modified on a `REGIONAL BY ROW` table within the same database.
```

File: _includes/v22.1/sql/indexes-regional-by-row.md [hunk 2]
```diff
@@ -1 +1,3 @@
  In [multi-region deployments](multiregion-overview.html), most users should use [`REGIONAL BY ROW` tables](multiregion-overview.html#regional-by-row-tables) instead of explicit index [partitioning](partitioning.html). When you add an index to a `REGIONAL BY ROW` table, it is automatically partitioned on the [`crdb_region` column](set-locality.html#crdb_region). Explicit index partitioning is not required.
+
+ While CockroachDB process an [`ADD REGION`](add-region.html) or [`DROP REGION`](drop-region.html) statement on a particular database, creating or modifying an index will throw an error. Similarly, all [`ADD REGION`](add-region.html) and [`DROP REGION`](drop-region.html) statements will be blocked an index is being modified on a `REGIONAL BY ROW` table within the same database.
```
Comment (gemma-shay) [inline _includes/v22.1/sql/indexes-regional-by-row.md:3]:
```suggestion
 While CockroachDB processes an [`ADD REGION`](add-region.html) or [`DROP REGION`](drop-region.html) statement on a particular database, creating or modifying an index will throw an error. Similarly, all [`ADD REGION`](add-region.html) and [`DROP REGION`](drop-region.html) statements will be blocked if an index is being modified on a `REGIONAL BY ROW` table within the same database.
```

File: v21.1/add-region.md [hunk 3]
```diff
@@ -4,7 +4,7 @@ summary: The ADD REGION statement adds a region to a multi-region database.
 toc: true
 ---
 
-<span class="version-tag">New in v21.1:</span> The `ALTER DATABASE .. ADD REGION` [statement](sql-statements.html) adds a [region](multiregion-overview.html#database-regions) to a [multi-region database](multiregion-overview.html).
+<span class="version-tag">New in v21.1:</span> The `ALTER DATABASE .. ADD REGION` [statement](sql-statements.html) adds a [region](multiregion-overview.html#database-regions) to a [multi-region database](multiregion-overview.html). While CockroachDB processes an index modification or changing a table to or from a [`REGIONAL BY ROW` table](multiregion-overview.html#regional-by-row-tables), attempting to drop a region from the database containing that `REGIONAL BY ROW` table will produce an error. Similarly, while this statement is running, all index modifications and locality changes on [`REGIONAL BY ROW`](multiregion-overview.html#regional-by-row-tables) tables will be blocked.
```
Comment (gemma-shay) [inline v21.1/add-region.md:7]:
```suggestion
<span class="version-tag">New in v21.1:</span> The `ALTER DATABASE .. ADD REGION` [statement](sql-statements.html) adds a [region](multiregion-overview.html#database-regions) to a [multi-region database](multiregion-overview.html). While CockroachDB is processing an index modification or changing a table to or from a [`REGIONAL BY ROW` table](multiregion-overview.html#regional-by-row-tables), attempting to drop a region from the database containing that `REGIONAL BY ROW` table will produce an error. Similarly, while this statement is running, all index modifications and locality changes on [`REGIONAL BY ROW`](multiregion-overview.html#regional-by-row-tables) tables will be blocked.
```

File: v21.2/drop-region.md [hunk 4]
```diff
@@ -5,7 +5,7 @@ toc: true
 docs_area: reference.sql
 ---
 
- The `ALTER DATABASE .. DROP REGION` [statement](sql-statements.html) drops a [region](multiregion-overview.html#database-regions) from a [multi-region database](multiregion-overview.html).
+The `ALTER DATABASE .. DROP REGION` [statement](sql-statements.html) drops a [region](multiregion-overview.html#database-regions) from a [multi-region database](multiregion-overview.html). While CockroachDB processes an index modification or changing a table to or from a [`REGIONAL BY ROW` table](multiregion-overview.html#regional-by-row-tables), attempting to drop a region from the database containing that `REGIONAL BY ROW` table will produce an error. Similarly, while this statement is running, all index modifications and locality changes on [`REGIONAL BY ROW`](multiregion-overview.html#regional-by-row-tables) tables will be blocked.
```
Comment (gemma-shay) [inline v21.2/drop-region.md:8]:
```suggestion
The `ALTER DATABASE .. DROP REGION` [statement](sql-statements.html) drops a [region](multiregion-overview.html#database-regions) from a [multi-region database](multiregion-overview.html). While CockroachDB is processing an index modification or changing a table to or from a [`REGIONAL BY ROW` table](multiregion-overview.html#regional-by-row-tables), attempting to drop a region from the database containing that `REGIONAL BY ROW` table will produce an error. Similarly, while this statement is running, all index modifications and locality changes on [`REGIONAL BY ROW`](multiregion-overview.html#regional-by-row-tables) tables will be blocked.
```

File: v22.1/add-region.md [hunk 5]
```diff
@@ -2,10 +2,10 @@
 title: ADD REGION
 summary: The ADD REGION statement adds a region to a multi-region database.
 toc: true
-docs_area: reference.sql 
+docs_area: reference.sql
 ---
 
- The `ALTER DATABASE .. ADD REGION` [statement](sql-statements.html) adds a [region](multiregion-overview.html#database-regions) to a [multi-region database](multiregion-overview.html).
+ The `ALTER DATABASE .. ADD REGION` [statement](sql-statements.html) adds a [region](multiregion-overview.html#database-regions) to a [multi-region database](multiregion-overview.html). While CockroachDB processes an index modification or changing a table to or from a [`REGIONAL BY ROW` table](multiregion-overview.html#regional-by-row-tables), attempting to drop a region from the database containing that `REGIONAL BY ROW` table will produce an error. Similarly, while this statement is running, all index modifications and locality changes on [`REGIONAL BY ROW`](multiregion-overview.html#regional-by-row-tables) tables will be blocked.
```
Comment (gemma-shay) [inline v22.1/add-region.md:8]:
```suggestion
 The `ALTER DATABASE .. ADD REGION` [statement](sql-statements.html) adds a [region](multiregion-overview.html#database-regions) to a [multi-region database](multiregion-overview.html). While CockroachDB is processing an index modification or changing a table to or from a [`REGIONAL BY ROW` table](multiregion-overview.html#regional-by-row-tables), attempting to drop a region from the database containing that `REGIONAL BY ROW` table will produce an error. Similarly, while this statement is running, all index modifications and locality changes on [`REGIONAL BY ROW`](multiregion-overview.html#regional-by-row-tables) tables will be blocked.
```

File: v22.1/set-locality.md [hunk 6]
```diff
@@ -5,7 +5,7 @@ toc: true
 docs_area: reference.sql
 ---
 
-The `ALTER TABLE .. SET LOCALITY` [statement](sql-statements.html) changes the [table locality](multiregion-overview.html#table-locality) of a [table](create-table.html) in a [multi-region database](multiregion-overview.html).
+The `ALTER TABLE .. SET LOCALITY` [statement](sql-statements.html) changes the [table locality](multiregion-overview.html#table-locality) of a [table](create-table.html) in a [multi-region database](multiregion-overview.html). While CockroachDB process an [`ADD REGION`](add-region.html) or [`DROP REGION`](drop-region.html) statement on a particular database, the `ALTER TABLE .. SET LOCALITY` will throw an error if enabling or disabling `REGIONAL BY ROW` on a table within that database. Similarly, all [`ADD REGION`](add-region.html) and [`DROP REGION`](drop-region.html) statements will be blocked while this query is running on a `REGIONAL BY ROW` table within the same database.
```
Comment (gemma-shay) [inline v22.1/set-locality.md:8]:
```suggestion
The `ALTER TABLE .. SET LOCALITY` [statement](sql-statements.html) changes the [table locality](multiregion-overview.html#table-locality) of a [table](create-table.html) in a [multi-region database](multiregion-overview.html). While CockroachDB is processing an [`ADD REGION`](add-region.html) or [`DROP REGION`](drop-region.html) statement on a particular database, the `ALTER TABLE .. SET LOCALITY` statement will throw an error if enabling or disabling `REGIONAL BY ROW` on a table within that database. Similarly, all [`ADD REGION`](add-region.html) and [`DROP REGION`](drop-region.html) statements will be blocked while this query is running on a `REGIONAL BY ROW` table within the same database.
```
Same comment as v21.1

File: v21.1/set-locality.md [hunk 7]
```diff
@@ -4,7 +4,7 @@ summary: The SET LOCALITY statement changes the locality of a table.
 toc: true
 ---
 
-<span class="version-tag">New in v21.1:</span> The `ALTER TABLE .. SET LOCALITY` [statement](sql-statements.html) changes the [table locality](multiregion-overview.html#table-locality) of a [table](create-table.html) in a [multi-region database](multiregion-overview.html).
+<span class="version-tag">New in v21.1:</span> The `ALTER TABLE .. SET LOCALITY` [statement](sql-statements.html) changes the [table locality](multiregion-overview.html#table-locality) of a [table](create-table.html) in a [multi-region database](multiregion-overview.html). While CockroachDB process an [`ADD REGION`](add-region.html) or [`DROP REGION`](drop-region.html) statement on a particular database, the `ALTER TABLE .. SET LOCALITY` will throw an error if enabling or disabling `REGIONAL BY ROW` on a table within that database. Similarly, all [`ADD REGION`](add-region.html) and [`DROP REGION`](drop-region.html) statements will be blocked while this query is running on a `REGIONAL BY ROW` table within the same database.
```
[the remark this answers is not in the document]
Comment (nickvigilante) [reply v21.1/set-locality.md]: I can tidy up the language here. Essentially, if there is an `ADD REGION` or `DROP REGION` statement running in the system at any given time and while that statement is running another user attempts to run `ALTER TABLE ... SET LOCALITY` at the same time, the `SET LOCALITY` statement will throw an error, but only if the locality change contained within that statement is enabling or disabling `REGIONAL BY ROW`.

[the remark this answers is not in the document]
Comment (gemma-shay) [reply v21.1/set-locality.md]: Ahhh I see. Okay how about this on a new line:
While CockroachDB is processing an `ALTER TABLE .. SET LOCALITY` statement that enables or disables `REGIONAL BY ROW` on a table within a database, any `ADD REGION` and `DROP REGION` statements on that database will fail.

[the remark this answers is not in the document]
Comment (nickvigilante) [reply v21.1/set-locality.md]: I like it. Pushing the changes now.

**Review by gemma-shay — APPROVED** [review]

**Comment by nickvigilante** [comment]
TFTR!

**Pull request merged**